### PR TITLE
Phase 1/color

### LINKED ISSUE #2: Phase 1: Website Body
# Body

home page

> This is a prototype of the homepage that includes the header, background, and footer.

The body of the page should be a black to red fade using the official colors of the University of Louisville.

## Requirements Checklist:

- [x] Body should include the title of the page currently selected

- [x] Background should be a black to red fade

CardTech website background

- [x] 50% black

- [x]  Blur between colors in the middle ( 15%)

- [x] 50% red

- [x] background will cover the entire page

- [x] Body should be part of an ASP.NET template, allowing other pages to insert their `<body>` content into the template.

### PR DESCRIPTION
**Pull request by Erika-Maglasang**
A made a branch that is a copy of the phase 1 footer pull request #21. 

Fixes #

- Applied the black to red fade on the body of the layout page, but the actual rule is located in the index page.

- The background image covers the whole page, thanks to the flex command.

- The background also has it's own stylesheet if we need to take it off of the index page.

Link to what the page looks like:
https://localhost:44367/

### CODE REVIEW

**Review by diagnose — CHANGES_REQUESTED** [review]
It’s typically best-practice to only create branches off of master. I will set up ASP.NET to support hosting css files and from there we can add in your code.

**Review by diagnose — APPROVED** [review]

**Comment by diagnose** [comment]
Code has been updated to use static files and apply across all pages. I give this the green light now. 👍🏼

**Review by Erika-Maglasang — COMMENTED** [review]
Link to what the website looks like after color changes have been applied:
https://localhost:44367/

LayoutColor contains all the stylesheet code.

**Comment by Erika-Maglasang** [comment]
> Link to what the website looks like after color changes have been applied:
> https://localhost:44367/
> 
> LayoutColor contains all the stylesheet code.

Screenshot 2020-10-16 00 05 20

File: CardTech.ResearchSite/CardTech.ResearchSite/Pages/Shared/_Layout.cshtml [hunk 1]
```diff
@@ -63,18 +63,18 @@
...
+    <footer class="footer mt-auto py-3">
         <div class="container-fluid">
             <div class="row align-items-center">
                 <div class="col-lg-auto border-right">
                     <img src="https://via.placeholder.com/350x100.png" />
                 </div>
                 <div class="col-lg-1 border-right">
                     <p>
-                        <a href="#">Login</a>
+                        <a class="footer-link"href="#">Login</a>
```
Comment (diagnose) [inline CardTech.ResearchSite/CardTech.ResearchSite/Pages/Shared/_Layout.cshtml]: I would remove `class="footer-link"`

File: CardTech.ResearchSite/CardTech.ResearchSite/Pages/Shared/_Layout.cshtml [hunk 2]
```diff
@@ -63,18 +63,18 @@
...
                     </p>
                     <p>
-                        <a href="#">Register</a>
+                        <a class="footer-link"href="#">Register</a>
```
Comment (diagnose) [inline CardTech.ResearchSite/CardTech.ResearchSite/Pages/Shared/_Layout.cshtml]: I would remove `class="footer-link"`

File: CardTech.ResearchSite/CardTech.ResearchSite/Pages/Shared/_Layout.cshtml [hunk 3]
```diff
@@ -83,22 +83,22 @@
                             <h5>Contact Us:</h5>
                             <p>
                                 Tel:
-                                <a href="tel:+15028526512">(502) 852-6512</a>
+                                <a class="footer-link"href="tel:+15028526512">(502) 852-6512</a>
```
Comment (diagnose) [inline CardTech.ResearchSite/CardTech.ResearchSite/Pages/Shared/_Layout.cshtml]: I would remove `class="footer-link"`

File: CardTech.ResearchSite/CardTech.ResearchSite/Pages/Shared/_Layout.cshtml [hunk 4]
```diff
@@ -83,22 +83,22 @@
...
                             </p>
                             <p>
                                 Fax:
-                                <a href="tel:+15028528361">(502) 852-8361</a>
+                                <a class="footer-link"href="tel:+15028528361">(502) 852-8361</a>
```
Comment (diagnose) [inline CardTech.ResearchSite/CardTech.ResearchSite/Pages/Shared/_Layout.cshtml]: I would remove `class="footer-link"`

File: CardTech.ResearchSite/CardTech.ResearchSite/Pages/Shared/_Layout.cshtml [hunk 5]
```diff
@@ -83,22 +83,22 @@
...
                             </p>
                         </div>
                         <div class="col-md-6">
                             <p>
-                                <a href="https://twitter.com/UofLInnovate" target="_blank">Twitter</a>
+                                <a class="footer-link"href="https://twitter.com/UofLInnovate" target="_blank">Twitter</a>
```
Comment (diagnose) [inline CardTech.ResearchSite/CardTech.ResearchSite/Pages/Shared/_Layout.cshtml]: I would remove `class="footer-link"`

File: CardTech.ResearchSite/CardTech.ResearchSite/Pages/Shared/_Layout.cshtml [hunk 6]
```diff
@@ -83,22 +83,22 @@
...
                             </p>
                             <p>
-                                <a href="https://www.facebook.com/UofLInnovate" target="_blank">Facebook</a>
+                                <a class="footer-link"href="https://www.facebook.com/UofLInnovate" target="_blank">Facebook</a>
```
Comment (diagnose) [inline CardTech.ResearchSite/CardTech.ResearchSite/Pages/Shared/_Layout.cshtml]: I would remove `class="footer-link"`

File: CardTech.ResearchSite/CardTech.ResearchSite/Pages/Shared/_Layout.cshtml [hunk 7]
```diff
@@ -83,22 +83,22 @@
...
                             </p>
                             <p>
-                                <a href="https://www.linkedin.com/school/uoflinnovate/" target="_blank">LinkedIn</a>
+                                <a class="footer-link"href="https://www.linkedin.com/school/uoflinnovate/" target="_blank">LinkedIn</a>
```
Comment (diagnose) [inline CardTech.ResearchSite/CardTech.ResearchSite/Pages/Shared/_Layout.cshtml]: I would remove `class="footer-link"`

File: CardTech.ResearchSite/CardTech.ResearchSite/Pages/Shared/_Layout.cshtml [hunk 8]
```diff
@@ -108,18 +108,18 @@
                     <div class="row">
                         <div class="col-md-6">
                             <p>
-                                <a href="#">Students</a>
+                                <a class="footer-link"href="#">Students</a>
```
Comment (diagnose) [inline CardTech.ResearchSite/CardTech.ResearchSite/Pages/Shared/_Layout.cshtml]: I would remove `class="footer-link"`

File: CardTech.ResearchSite/CardTech.ResearchSite/Pages/Shared/_Layout.cshtml [hunk 9]
```diff
@@ -108,18 +108,18 @@
...
                             </p>
                             <p>
-                                <a href="#">Community</a>
+                                <a class="footer-link" href="#">Community</a>
```
Comment (diagnose) [inline CardTech.ResearchSite/CardTech.ResearchSite/Pages/Shared/_Layout.cshtml]: I would remove `class="footer-link"`

File: CardTech.ResearchSite/CardTech.ResearchSite/Pages/Shared/_Layout.cshtml [hunk 10]
```diff
@@ -108,18 +108,18 @@
...
                             </p>
                         </div>
                         <div class="col-md-6">
                             <p>
-                                <a asp-page="/Industry">Industry</a>
+                                <a class="footer-link" asp-page="/Industry">Industry</a>
```
Comment (diagnose) [inline CardTech.ResearchSite/CardTech.ResearchSite/Pages/Shared/_Layout.cshtml]: I would remove `class="footer-link"`

File: CardTech.ResearchSite/CardTech.ResearchSite/Pages/Shared/_Layout.cshtml [hunk 11]
```diff
@@ -108,18 +108,18 @@
...
                             </p>
                             <p>
-                                <a asp-page="/Research">Researchers</a>
+                                <a class="footer-link" asp-page="/Research">Researchers</a>
```
Comment (diagnose) [inline CardTech.ResearchSite/CardTech.ResearchSite/Pages/Shared/_Layout.cshtml]: I would remove `class="footer-link"`

File: CardTech.ResearchSite/CardTech.ResearchSite/wwwroot/css/LayoutColor.css [hunk 12]
```diff
@@ -3,4 +3,16 @@
...
+    background-color: rgb(173, 0, 0);
+    border-bottom: black;
+    color:white;
+}
+.footer {
+    background-color: rgb(173, 0, 0);
+    border-top:black;
+    color:white;
+}
+.nav, .nav-item, .nav-link, .footer-item, .footer-link{
```
Comment (diagnose) [inline CardTech.ResearchSite/CardTech.ResearchSite/wwwroot/css/LayoutColor.css]: You should add in a class that reads

`
.footer a {
    color: white;
}
`

Which will remove the need for adding individual classes to each link.

**Review by diagnose — CHANGES_REQUESTED** [review]
Please see my comments regarding the footer.

File: CardTech.ResearchSite/CardTech.ResearchSite/Pages/Shared/_Layout.cshtml [hunk 13]
```diff
@@ -71,10 +71,10 @@
                 </div>
                 <div class="col-lg-1 border-right">
                     <p>
-                        <a class="footer-link"href="#">Login</a>
+                        <a class="footer"href="#">Login</a>
```
Comment (diagnose) [inline CardTech.ResearchSite/CardTech.ResearchSite/Pages/Shared/_Layout.cshtml]: Sorry if I wasn't clear, but there shouldn't be **any** class on these tags.

**Comment by Erika-Maglasang** [comment]
I took away the footer-link class and replaced it with the existing footer class. At first I tried to apply the class to the whole footer, but it only applied it to the header text. Would it be better to place it in the div, p, or the col-lg-1 - 6? Is there an easier way to override the default blue link color?

**Comment by Erika-Maglasang** [comment]
Screenshot 2020-10-16 14 05 07

All links on the footer will be left as the default blue for the time being.

**Comment by diagnose** [comment]
> I took away the footer-link class and replaced it with the existing footer class. At first I tried to apply the class to the whole footer, but it only applied it to the header text. Would it be better to place it in the div, p, or the col-lg-1 - 6? Is there an easier way to override the default blue link color?

The fix was:

`
footer a {
color: white;
}`

I've committed this change.

**Review by GMStull — APPROVED** [review]
Looks solid to me, no issues.

**Comment by diagnose** [comment]
Two issues I've noticed in mobile:

1. The button that normally appears on the header is now missing.

2. The rounded corners looks ugly on mobile. We should probably use an entirely white background in this environment.

image

**Review by JCDILL03 — APPROVED** [review]
The code looks good to me... especially everything with the colors and columns

File: CardTech.ResearchSite/CardTech.ResearchSite/Pages/About.cshtml [hunk 14]
```diff
@@ -6,3 +6,29 @@
 
 <h1>About</h1>
 
+<body>
```
Comment (diagnose) [inline CardTech.ResearchSite/CardTech.ResearchSite/Pages/About.cshtml]: I'm having trouble understanding this... is there a reason we have some sample code here?

File: CardTech.ResearchSite/CardTech.ResearchSite/wwwroot/css/LayoutColor.css [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+.navbar, .nav, .nav-item, .nav-link, .footer {
+    color: white;
+}
+
+.main-content {
+    margin: 2.5% auto;
+    border-radius: 7px;
+}
+/* Container for flexboxes */
+.row {
```
Comment (diagnose) [inline CardTech.ResearchSite/CardTech.ResearchSite/wwwroot/css/LayoutColor.css]: These are already classes in bootstrap. Is this relevant/does it interfere?

**Review by diagnose — CHANGES_REQUESTED** [review]
Little confused by the commit made by @GMStull. Please see the comments I left.

**Review by diagnose — APPROVED** [review]
Good for now, will make additional changes in `enhancement/*` branches as needed. Merging.

**Pull request merged**